core/conversion/evaluators/prim.cpp

```diff
@@ -48,7 +48,13 @@ auto prim_registrations =
                           c10::List<int64_t> list;
                           list.reserve(num_inputs);
                           for (auto in : n->inputs()) {
-                            list.emplace_back(std::move(args.at(in).unwrapToInt()));
+                            if (args.at(in).IValue()->isInt()) {
+                              list.emplace_back(std::move(args.at(in).unwrapToInt()));
+                            } else if (args.at(in).IValue()->isTuple()) {
```

Comment:
Is this valid in TorchScript? like I guess this would be:

```
%a : (Int) = prim::Constant[value=(1,2,3)]
%b : (Int) = prim::Constant[value=(4,5,6)]
%x : Int[] = prim::ListConstruct(%a, %b)
```
Where `%x` is `[1,2,3,4,5,6]` or something? 

Wouldn't this also let you do: 
```
%a : (Int) = prim::Constant[value=(1,2,3)]
%b : Int = prim::Constant[value=4]
%x : Int[] = prim::ListConstruct(%a, %b)
```

Where `%x` would be `[1, 2, 3, 4]`

Reply:
Naively I would think that the first case should result in `[(1,2,3),(4,5,6)]` and the second case shouldnt be allowed

Reply:
That stemmed out from my early attempts to use rewriting rules where I followed some existing rewriting sample - so apparently it is valid. If there are any doubts - we do not have to include this fix, as I ended up not doing rewrites.